packages/strapi-plugin-content-type-builder/controllers/validation/types.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const yup = require('yup');
+const { validators, VALID_TYPES, isValidName } = require('./common');
+
+module.exports = obj => {
+  return {
+    type: yup
+      .string()
+      .oneOf(VALID_TYPES)
+      .required(),
+    ...getTypeShape(obj),
+  };
+};
+
+const getTypeShape = obj => {
+  switch (obj.type) {
+    /**
+     * complexe types
+     */
+
+    case 'media': {
+      return {
+        multiple: yup.boolean(),
+        required: validators.required,
+        unique: validators.unique,
+      };
+    }
+
+    /**
+     * scalar types
+     */
+    case 'string':
+    case 'text':
+    case 'richtext': {
+      return {
+        default: yup.string(),
+        required: validators.required,
+        unique: validators.unique,
+        min: validators.minLength,
+        max: validators.maxLength,
+      };
+    }
+    case 'json': {
+      return {
+        required: validators.required,
+        unique: validators.unique,
+      };
+    }
+    case 'enumeration': {
+      return {
+        enum: yup
+          .array()
+          .of(yup.string().test(isValidName))
+          .min(1)
+          .required(),
+        default: yup
+          .string()
+          .when('enum', enumVal => yup.string().oneOf(enumVal)),
+        enumName: yup.string().test(isValidName),
+        required: validators.required,
+        unique: validators.unique,
+      };
+    }
+    case 'password': {
+      return {
+        required: validators.required,
+        min: validators.minLength,
+        max: validators.maxLength,
+      };
+    }
+    case 'email': {
+      return {
+        default: yup.string().email(),
+        required: validators.required,
+        unique: validators.unique,
+        min: validators.minLength,
+        max: validators.maxLength,
+      };
+    }
+    case 'integer': {
+      return {
+        default: yup.number().integer(),
+        required: validators.required,
+        unique: validators.unique,
+        min: yup
+          .number()
+          .integer()
+          .positive(),
```

Comment:
Does this mean that we can't set the minimum value of integer to `-2` ?
Same for the maximum

Reply:
Indeed changed it